src/main/webapp/resources/js/contexts/AnalysisDetailsContext.js

```diff
@@ -0,0 +1,131 @@
+import React, { useReducer, useContext } from "react";
+
+import {
+  getVariablesForDetails,
+  getAnalysisInputFiles,
+  saveToRelatedSamples,
+  updateAnalysisEmailPipelineResult,
+  updateAnalysis
+} from "../apis/analysis/analysis";
+
+import { showNotification } from "../modules/notifications";
+import { AnalysisContext } from "../contexts/AnalysisContext";
+
+const TYPES = {
+  DETAILS: "ANALYSIS_DETAILS",
+  EMAIL_PIPELINE_RESULT: "UPDATED_EMAIL_PIPELINE_RESULT",
+  PRIORITY: "UPDATED_PRIORITY",
+  SAMPLES: "SAMPLES_DATA",
+  UPDATE_SAMPLES: "UPDATE_SAMPLES"
+};
+
+const reducer = (context, action) => {
+  switch (action.type) {
+    case TYPES.DETAILS:
+      return {
+        ...context,
+        ...action.payload
+      };
+    case TYPES.EMAIL_PIPELINE_RESULT:
+      return { ...context, emailPipelineResult: action.emailPipelineResult };
+    case TYPES.PRIORITY:
+      return { ...context, priority: action.priority };
+    case TYPES.UPDATE_SAMPLES:
+      return { ...context, updateSamples: action.updateSamples };
+    case TYPES.SAMPLES:
+      return {
+        ...context,
+        ...action.payload
+      };
+    default:
+      return;
+  }
+};
+
+const initialContext = {
+  emailPipelineResult: window.PAGE.analysis.emailPipelineResult,
+  workflowName: null,
+  version: null,
+  duration: null,
+  createdDate: null,
+  priority: null,
+  priorities: [],
+  canShareToSamples: false,
+  updatePermission: false,
+  samples: [],
+  sequenceFilePairList: [],
+  sequenceFileSizeList: [],
+  referenceFile: [],
+  updateSamples: window.PAGE.analysis.updateSamples
```

Comment:
Why are you adding these to the page?  Would it not make more sense (and provide a faster page load) if you were to load those after page load?

Reply:
Completely forgot to update these when I was refactoring. Will be fixed in next commit.

Reply:
Fixed in f8844707a5f60a885ba87a349edb62b076eaba15